### ISSUE TITLE
Template K8s guest os ubuntu failed / resolv failed

### ISSUE THREAD

**Issue by defmond972** [vmware/container-service-extension#432]
Hello,

When CSE is configuring templates K8s, it works for phontos OS but for ubuntu it failed. 

I got the following error: Could not resolve host: download.docker.com or Temporary failure resolving 'security.ubuntu.com'.

On console vm temp, i don't have ip, but after a restart for networking service, i get an ip and could resolve url. 

The network connected to the vm temp is working , because the template based on phontos os works for templating.

I find that we have a bug, am i wrong ? 

Best regards

**Comment by defmond972** [comment]
I resolved my issue; i adding in the cust.sh the following line :
systemctl restart networking.service
while [ `systemctl is-active networking` != 'active' ]; do echo 'waiting for network'; sleep 5; done
after ==> growpart /dev/sda 1 || :
resize2fs /dev/sda1 || :

**Comment by goelaashima** [comment]
Which version of CSE were/are you using?

**Comment by defmond972** [comment]
hi, 2.5.0.0b1

**Comment by andrew-ni** [comment]
Hi @defmond972 , are you still experiencing issues with this? It seems like this is just a network issue where the VM was not able to connect to those hosts.

What exactly did you add to your customization script?

**Comment by defmond972** [comment]
Hi, Andrew, no more now but i needed to add a fix, i added in the cust file "cust.sh" for the template, a new reload for the network service. 
Example: 

systemctl restart networking.service
while [ systemctl is-active networking != 'active' ]; do echo 'waiting for network'; sleep 5; done

after ==> growpart /dev/sda 1 || :
resize2fs /dev/sda1 || :

**Comment by fhufenreuter** [comment]
Hi,
i do have the same issue as @defmond972. I also used 2.5.0.b1 and it is still the same in 2.5.0 GA
The workaround mentioned works for me as well

**Comment by tarmon** [comment]
Here's the issue as its happening.

---
Creating vApp 'ubuntu-16.04_k8-1.13_weave-2.3.0_temp'
Created vApp 'ubuntu-16.04_k8-1.13_weave-2.3.0_temp'
Customizing vApp 'ubuntu-16.04_k8-1.13_weave-2.3.0_temp', vm 'Temp-vm'
Found data file: /root/.cse_scripts/ubuntu-16.04_k8-1.13_weave-2.3.0_rev1/cust.sh
Rebooting vApp 'ubuntu-16.04_k8-1.13_weave-2.3.0_temp'
---

Basically within 10 seconds of the VM being powered on, the installer attempts to issue a reboot for some reason, which fails because the tools are not yet running. The VM hangs for an extended period before the tools become available (minutes) - I'm guessing it doesn't like not having DHCP and its a network hang that happens before the tools load. 

So basically, the customization is delayed for long enough that the installer starts waiting for the tools to become available to start the rest of the config, without the reboot that presumably was put in place to start the networking up properly not actually happening. The rest of the install then fails due to lack of network connectivity.

**Comment by andrew-ni** [comment]
@defmond972 @flamby, is the fix really the `after ==>` part? I'm trying to incorporate this fix, but I keep getting `after: command not found`, and I can't find any online resources talking about `after ==>`.

Could I see the actual file change that you guys made? Or can you guys give some insight on using `after ==>` and what it does?

**Comment by andrew-ni** [comment]
@tarmon Based on our understanding/testing for customizing VMs in this vCD/vCenter environment, we have to reboot the VM in order for the tools to start up and run properly. Once the reboot finishes, then we wait for the tools to be ready and proceed with customization.

Is your understanding of this different? Can you elaborate a little more?

**Comment by flamby** [comment]
@andrew-ni 

here is the patch

```
systemctl restart networking.service
while [ `systemctl is-active networking` != 'active' ]; do echo 'waiting for network'; sleep 5; done

growpart /dev/sda 1 || :
resize2fs /dev/sda1 || :

+systemctl restart networking.service
+while [ `systemctl is-active networking` != 'active' ]; do echo 'waiting for network'; sleep 5; done
```
So as you see, we simply copy/paste the same two lines after the partitionning.
I guess the later needs time to sync, thus the restart of the network and the sleep avoid a timeout.
Simple as that.

**Comment by andrew-ni** [comment]
@flamby thanks! Will test it out and likely implement this in our templates
